Framework/PythonInterface/test/python/mantid/FunctionWrapperTest.py

```diff
@@ -0,0 +1,376 @@
+        self.assertEqual(fc3_str.count("Sigma="),3)
+        self.assertEqual(fc3_str.count("Sigma=1.25"),1)
+        self.assertEqual(fc3_str.count("Sigma=1.3"),1)
+        
+    def test_add(self):
+        g0 = FunctionWrapper( "Gaussian", Height=7.5, Sigma=1.2, PeakCentre=10)  
+        g1 = FunctionWrapper( "Gaussian", Height=8.5, Sigma=1.25, PeakCentre=12)  
+        lb = FunctionWrapper("LinearBackground")
+        
+        c = lb + g0 + g1
+        
+        c_str = c.__str__()
+        self.assertEqual(c_str.count("("),0)
+        self.assertEqual(c_str.count("LinearBackground"),1)
+        self.assertEqual(c_str.count("Gaussian"),2)
+        
+        lb_str = lb.__str__()
+        c0_str = c[0].__str__()
+        self.assertEqual(c0_str, lb_str)
+           
+        g0_str = g0.__str__()
+        c1_str = c[1].__str__()
+        self.assertEqual(c1_str, g0_str)
+        
+        g1_str = g1.__str__()
+        c2_str = c[2].__str__()
+        self.assertEqual(c2_str, g1_str)
+        
+    def test_productfunction_creation(self):
+        g0 = FunctionWrapper( "Gaussian", Height=7.5, Sigma=1.2, PeakCentre=10)
+        g1 = FunctionWrapper( "Gaussian", Height=8.5, Sigma=1.2, PeakCentre=11)
+        testhelpers.assertRaisesNothing(self, ProductFunctionWrapper, g0, g1)
+
+    def test_mul(self):
+        g0 = FunctionWrapper( "Gaussian", Height=7.5, Sigma=1.2, PeakCentre=10)  
+        g1 = FunctionWrapper( "Gaussian", Height=8.5, Sigma=1.25, PeakCentre=12)  
+        lb = FunctionWrapper("LinearBackground")
+        
+        p = lb * g0 * g1
+        
+        p_str = p.__str__()
+        self.assertEqual(p_str.count("("),0)
+        self.assertEqual(p_str.count("LinearBackground"),1)
+        self.assertEqual(p_str.count("Gaussian"),2)
+        
+        lb_str = lb.__str__()
+        p0_str = p[0].__str__()
+        self.assertEqual(p0_str, lb_str)
+           
+        g0_str = g0.__str__()
+        p1_str = p[1].__str__()
+        self.assertEqual(p1_str, g0_str)
+        
+        g1_str = g1.__str__()
+        p2_str = p[2].__str__()
+        self.assertEqual(p2_str, g1_str)   
+
+    def test_convolution_creation(self):
+        g0 = FunctionWrapper( "Gaussian", Height=7.5, Sigma=1.2, PeakCentre=10)
+        g1 = FunctionWrapper( "Gaussian", Height=8.5, Sigma=1.2, PeakCentre=11)
+        testhelpers.assertRaisesNothing(self, ConvolutionWrapper, g0, g1)
+        
+    def test_multidomainfunction_creation(self):
+        g0 = FunctionWrapper( "Gaussian", Height=7.5, Sigma=1.2, PeakCentre=10)
+        g1 = FunctionWrapper( "Gaussian", Height=8.5, Sigma=1.2, PeakCentre=11)
+        testhelpers.assertRaisesNothing(self, MultiDomainFunctionWrapper, g0, g1)
+        m = MultiDomainFunctionWrapper( g0, g1)
+        self.assertEqual( m.nDomains(), 2)
+        
+    def test_prefinedfunction(self):
+        testhelpers.assertRaisesNothing(self, Gaussian, Height=7.5, Sigma=1.2, PeakCentre=10)
+       
```

Comment:
Test that the wrappers produce correct results when used in `EvaluateFunction` algorithm.